protocol/x/prices/keeper/slinky_adapter.go

```diff
@@ -49,10 +45,10 @@ func (k Keeper) GetCurrencyPairFromID(ctx sdk.Context, id uint64) (cp slinkytype
 }
 
 func (k Keeper) GetIDForCurrencyPair(ctx sdk.Context, cp slinkytypes.CurrencyPair) (uint64, bool) {
-	// check in the keeper's cache first
-	id, found := k.currencyPairIDCache.GetIDForCurrencyPair(cp.String())
+	// Try to get corresponding market ID of the currency pair from the store
+	marketId, found := k.GetCurrencyPairIDFromStore(ctx, cp)
 	if found {
-		return id, true
+		return uint64(marketId), true
 	}
 
 	// if not found, iterate through all market params and find the id
```

Comment:
For ease of reasoning, think we should maintain the invariant that the cache is always well-populated, instead of relying on a lazy approach which made more sense when the cache was transient. 

This means:
-  Explicitly setting up the cache for all markets in v7 upgrade handler. 
- No need to maintain the "if not found" case